internal/cmd/egctl/version.go

```diff
@@ -0,0 +1,126 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package egctl
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/yaml"
+
+	"github.com/envoyproxy/gateway/internal/cmd/options"
+	"github.com/envoyproxy/gateway/internal/cmd/version"
+	kube "github.com/envoyproxy/gateway/internal/kubernetes"
+)
+
+const (
+	yamlOutput = "yaml"
+)
+
+func NewVersionsCommand() *cobra.Command {
+	var (
+		output          string
+		egContainerName string
+	)
+
+	versionCommand := &cobra.Command{
+		Use:     "version",
+		Aliases: []string{"versions", "v"},
+		Short:   "Show versions",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return versions(cmd.OutOrStdout(), egContainerName, output)
+		},
+	}
+
+	flags := versionCommand.Flags()
+	options.AddKubeConfigFlags(flags)
+
+	versionCommand.PersistentFlags().StringVarP(&output, "output", "o", yamlOutput, "One of 'yaml' or 'json'")
+	versionCommand.PersistentFlags().StringVar(&egContainerName, "eg-container-name", "envoy-gateway", "Name of the Envoy Gateway container")
```

Comment:
I don't expect the EG container name to change so I'm not sure why the container name needs to be exposed to a user. However, I can see EG being run in a different ns than `envoy-gateway-system` and expose the ns to the user while defaulting to "envoy-gateway-system".

Reply:
`version` will search all namespace, expect vendors may use EG multi-tenancy with custom container name？

Reply:
I'm not following your point here about the EG container name. For multitenancy, a user would install EG in different namespaces with each EG using a different `controllerName`. A user would then create a GatewayClass per EG and one or more Gateways that reference the GatewayClass. By default, egctl should expect EG in the `envoy-gateway-system` namespace with the Deployment name of `envoy-gateway`. If egctl needs to communicate with an EG in a different namespace, a flag such as `--namespace` should be surfaced. For security purposes, we should avoid providing egctl access across all namespaces.

Reply:
let me remove this part first.